ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiSnapshotWorldState.java

```diff
@@ -27,21 +28,26 @@
  * non-persisting mutable world state rather than writing worldstate changes directly.
  */
 public class BonsaiSnapshotWorldState extends BonsaiPersistedWorldState
-    implements SnapshotMutableWorldState {
+    implements SnapshotMutableWorldState, BonsaiStorageSubscriber {
 
   private final SnappedKeyValueStorage accountSnap;
   private final SnappedKeyValueStorage codeSnap;
   private final SnappedKeyValueStorage storageSnap;
   private final SnappedKeyValueStorage trieBranchSnap;
+  private final BonsaiWorldStateKeyValueStorage parentWorldStateStorage;
+  private final Long parentWorldStateSubscriberId;
 
   private BonsaiSnapshotWorldState(
       final BonsaiWorldStateArchive archive,
-      final BonsaiSnapshotWorldStateKeyValueStorage snapshotWorldStateStorage) {
+      final BonsaiSnapshotWorldStateKeyValueStorage snapshotWorldStateStorage,
+      final BonsaiWorldStateKeyValueStorage parentWorldStateStorage) {
     super(archive, snapshotWorldStateStorage);
     this.accountSnap = (SnappedKeyValueStorage) snapshotWorldStateStorage.accountStorage;
     this.codeSnap = (SnappedKeyValueStorage) snapshotWorldStateStorage.codeStorage;
     this.storageSnap = (SnappedKeyValueStorage) snapshotWorldStateStorage.storageStorage;
     this.trieBranchSnap = (SnappedKeyValueStorage) snapshotWorldStateStorage.trieBranchStorage;
+    this.parentWorldStateStorage = parentWorldStateStorage;
+    parentWorldStateSubscriberId = parentWorldStateStorage.subscribe(this);
```

Comment:
and are we sure we manage all the case when the subscribe throw a runtime exception ? I'm not sure there are a lot of case

Reply:
I moved the subscribe into a fluent member and added it to the static create call :+1:

Re: runtime exceptions, I don't see much opportunity for that to happen within .subscribe(), but I wrapped it in a try-catch just in case